server/src/main/java/org/elasticsearch/search/aggregations/bucket/geogrid/GeoGridAggregationBuilder.java

```diff
@@ -101,20 +146,43 @@ protected AggregationBuilder shallowCopy(Builder factoriesBuilder, Map<String, O
      */
     public GeoGridAggregationBuilder(StreamInput in) throws IOException {
         super(in, ValuesSourceType.GEOPOINT, ValueType.GEOPOINT);
+        type = GeoHashType.readFromStream(in);
         precision = in.readVInt();
         requiredSize = in.readVInt();
         shardSize = in.readVInt();
     }
 
     @Override
     protected void innerWriteTo(StreamOutput out) throws IOException {
+        type.writeTo(out);
```

Comment:
We need to read the version form the stream here and only write this to the stream if the version is on or after the version we add this feature.